File: lib/go/telepact/internal/types/TError.go
//|
//|  Copyright The Telepact Authors
//|
//|  Licensed under the Apache License, Version 2.0 (the "License");
//|  you may not use this file except in compliance with the License.
//|  You may obtain a copy of the License at
//|
//|  https://www.apache.org/licenses/LICENSE-2.0
//|
//|  Unless required by applicable law or agreed to in writing, software
//|  distributed under the License is distributed on an "AS IS" BASIS,
//|  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//|  See the License for the specific language governing permissions and
//|  limitations under the License.
//|

package types

// TError represents an error type wrapping an underlying union.
type TError struct {
	Name   string
	Errors *TUnion
}

// NewTError constructs a TError instance.
func NewTError(name string, errors *TUnion) *TError {
	return &TError{Name: name, Errors: errors}
}

File: lib/py/telepact/internal/types/TError.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .TUnion import TUnion


class TError:
    def __init__(self, name: str, errors: 'TUnion'):
        self.name = name
        self.errors = errors
